src/DependencyInjection/CompilerPass/InheritancePass.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the CsaGuzzleBundle package
+ *
+ * (c) Charles Sarrazin <charles@sarraz.in>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code
+ */
+
+namespace Csa\Bundle\GuzzleBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+
+/**
+ * Csa Guzzle definition inheritancePass.
+ *
+ * @author Sofiane HADDAG <sofiane.haddag@yahoo.fr>
+ */
+class InheritancePass implements CompilerPassInterface
+{
+    const TAG = 'csa_guzzle.inheritance';
+
+    private $checkedNodes;
```

Comment:
The same logic is used for https://github.com/SofHad/symfony/blob/master/src/Symfony/Component/DependencyInjection/Compiler/CheckCircularReferencesPass.php#L30